sql/moz-fx-data-shared-prod/telemetry_derived/newtab_desktop_daily_engagement_by_tile_id_position_v1/query.sql

```diff
@@ -0,0 +1,165 @@
+    tile_id,
+    position
+),
+  -- GLEAN Query
+glean_deduplicated_pings AS (
+  SELECT
+    submission_timestamp,
+    document_id,
+    normalized_country_code,
+    client_info,
+    events
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_live.newtab_v1`
+  WHERE
+    DATE(submission_timestamp) >= '2024-11-15'
+    AND DATE(submission_timestamp) = @submission_date
+  QUALIFY
+    ROW_NUMBER() OVER (
+      PARTITION BY
+        DATE(submission_timestamp),
+        document_id
+      ORDER BY
+        submission_timestamp DESC
+    ) = 1
+),
+glean_flattened_pocket_events AS (
+  SELECT
+    document_id,
+    submission_timestamp,
+    events.name AS event_name,
+    mozfun.map.get_key(events.extra, 'recommendation_id') AS recommendation_id,
+    mozfun.map.get_key(events.extra, 'tile_id') AS tile_id,
+    mozfun.map.get_key(events.extra, 'position') AS position,
+    COUNT(1) OVER (PARTITION BY document_id, events.name) AS user_event_count
+  FROM
+    glean_deduplicated_pings,
+    UNNEST(events) AS events
+  WHERE
+    events.category = 'pocket'
+    AND events.name IN ('impression', 'click', 'save', 'dismiss')
+    AND (
+      mozfun.map.get_key(events.extra, 'recommendation_id') IS NOT NULL
+      OR mozfun.map.get_key(events.extra, 'tile_id') IS NOT NULL
+    )
+    AND mozfun.map.get_key(events.extra, 'scheduled_corpus_item_id') IS NULL
+    AND SAFE_CAST(SPLIT(client_info.app_display_version, '.')[0] AS int64) >= 121
+),
+glean_summary AS (
+  SELECT
+    submission_timestamp,
+    recommendation_id,
+    COALESCE(SAFE_CAST(tile_id AS int), -1) AS tile_id,
+    COALESCE(SAFE_CAST(position AS int), -1) AS position,
+    SUM(CASE WHEN event_name = 'impression' THEN 1 ELSE 0 END) AS impression_count,
+    SUM(CASE WHEN event_name = 'click' THEN 1 ELSE 0 END) AS click_count,
+    SUM(CASE WHEN event_name = 'save' THEN 1 ELSE 0 END) AS save_count,
+    SUM(CASE WHEN event_name = 'dismiss' THEN 1 ELSE 0 END) AS dismiss_count
+  FROM
+    glean_flattened_pocket_events
+  WHERE
+    NOT (user_event_count > 50 AND event_name = 'click')
+  GROUP BY
+    submission_timestamp,
+    recommendation_id,
+    tile_id,
+    position
+)
+-- Combined Query with Full Outer Join
+SELECT
+  COALESCE(l.submission_timestamp, g.submission_timestamp) AS submission_timestamp,
+  COALESCE(l.recommendation_id, g.recommendation_id) AS recommendation_id,
+  COALESCE(l.tile_id, g.tile_id) AS tile_id,
+  COALESCE(l.position, g.position) AS position,
+  IFNULL(l.impression_count, 0) + IFNULL(g.impression_count, 0) AS total_impression_count,
+  IFNULL(l.click_count, 0) + IFNULL(g.click_count, 0) AS total_click_count,
+  IFNULL(l.save_count, 0) + IFNULL(g.save_count, 0) AS total_save_count,
+  IFNULL(l.dismiss_count, 0) + IFNULL(g.dismiss_count, 0) AS total_dismiss_count
+FROM
+  legacy_summary AS l
+FULL OUTER JOIN
+  glean_summary AS g
+  ON l.submission_timestamp = g.submission_timestamp
+  AND l.tile_id = g.tile_id
+  AND l.position = g.position
```

Comment:
I don't know about this. I think this ought to be a UNION ALL, not a join. Aren't we trying to capture all behavior from both clients on old versions (legacy telemetry) and newer versions (Glean)?